article/_posts/2020-10-15-vision_transformer.md

```diff
@@ -0,0 +1,80 @@
+* The usefulness/superiority of the proposed transformer over CNNs only appears when the number of images in the dataset reaches about 100 million
+* The proposed architecture uses about 2X less compute to train, compared to CNNs, for a given performance
+* The authors have used more than 25,000 TPUv3 * days over their experiments
+
+# Methods
+
+![](/article/images/vit/fig1.jpg)
+
+## Architecture
+
+* Images are split into patches (16x16 yields the best results)
+* The patches are flattened, and become the tokens (or "words")
+* The flattened patches are projected using a MLP
+* Each flattened patch is concatenated with a vector which represents the position of the patch. A positional embedding is learned, which maps a one-dimensional patch index to a vector representation.
+* Note that this allows the very first layer to attend to any part of the image, in comparison with CNNs for which the receptive field develops over many layers.
+* The first token is a dummy, and is only there because the feature vector of the image will appear at that position at the last layer of the network.
+
+## Experiment design
+
+* The models are pre-trained on either one of ImageNet (1k classes, 1.3M images), ImageNet-21k (21k classes, 14M images) or JFT (18k classes, 303M images).
+* The models are then fined-tuned on one of the datasets listed in the "Benchmarking datasets" section below.
+* Sometimes, the model is not fine tuned, but is evaluated in a few-shot regime. This is not well described in the paper: "Few-shot accuracies are obtained by solving a regularized linear regression problem that maps the (frozen) representation of a subset of training images to $$\{−1,1\}^K$$ target vectors. Though we mainly focus on fine-tuning performance, we sometimes use linear few-shot accuracies for fast on-the-fly evaluation where fine-tuning would be too costly."
+* They also have tried self-supervised pre-training, in which the model predicts masked patches. This is given little importance, and the results are "only" promising, so I will not write about this further.
+
+## Benchmarking datasets
+
+The experiments have been run on a number of datasets of image classification:
+
+* ImageNet
+* ImageNet ReaL ("Reassessed Labels", from Beyer et al. 2020) [Code](https://github.com/google-research/reassessed-imagenet)
+* CIFAR10/100
+* Oxford Pets, Oxford Flowers
+* VTAB (Zhai et al., 2019b) ("VTAB evaluates low-data transfer using 1 000 examples to diverse tasks.  The tasks are divided into three groups: Natural– tasks like the above, Pets, CIFAR, etc. Specialized– medical and satellite imagery, and Structured– tasks that require geometric understanding like localization.") [Blog post](https://ai.googleblog.com/2019/11/the-visual-task-adaptation-benchmark.html)
+
+# Results
+
+A seen in the table below, ViT performs slightly better to a very large ResNet, and does so using significantly less (fine-tuning) compute.
```

Comment:
using significantly less (fine-tuning) computational resources?

Not sure what the fine-tuning explanation means in here/how it relates to the computational resources.

Reply:
There are two phases: pre-training and fine-tuning. Both phases requires different amounts of FLOPS